cupy/cuda/memory.pyx

```diff
@@ -43,6 +47,52 @@ cdef class Memory:
         return self.ptr
 
 
+cdef class ManagedMemory(Memory):
+
+    """Memory allocation on a CUDA device.
+
+    This class provides an RAII interface of the CUDA memory allocation.
+
+    Args:
+        device (cupy.cuda.Device): Device whose memory the pointer refers to.
+        size (int): Size of the memory allocation in bytes.
+
+    """
+
+    def __init__(self, Py_ssize_t size):
+        self.size = size
+        self.device = None
+        self.ptr = 0
+        if size > 0:
+            self.device = device_mod.Device()
+            self.ptr = runtime.mallocManaged(size)
+            self._support_prefetch_advise = (
+                _cuda_version >= 8000 and
+                int(self.device.compute_capability) >= 60)
+
+    cpdef prefetch(self, stream):
+        """ Prefetch.
+
+        Args:
+            stream (cupy.cuda.Stream): Stream
+
+        """
+        if self._support_prefetch_advise:
+            runtime.memPrefetchAsync(self.ptr, self.size, self.device.id,
+                                     stream.ptr)
+
+    cpdef advise(self, int advise, device_mod.Device device):
+        """ Advise about the usage of this memory.
+
+        Args:
+            advics (int): Advise to be applied for this memory.
+            device (cupy.cuda.Device): Device to apply the advice for.
+
+        """
+        if self._support_prefetch_advise:
```

Comment:
Do we need to show warning? Users cannot know that this object ignores this method.